variables.tf

```diff
@@ -1,11 +1,154 @@
-##############################################################################
-# Input Variables
-##############################################################################
+variable "resource_group_id" {
+  description = "ID of resource group to use when creating the event stream instance"
+  type        = string
+}
 
-#variable "myvariable" {
-#  type        = string
-#  description = "A description of my variable"
-#  default     = ""
-#}
+variable "es_name" {
+  type        = string
+  description = "Name of the resource instance"
+}
 
-##############################################################################
+variable "plan" {
+  type        = string
+  description = "Plan for the event streams instance : lite, standard or enterprise-3nodes-2tb"
+  default     = null
+  validation {
+    condition     = contains(["lite", "standard", "enterprise-3nodes-2tb"], var.plan)
+    error_message = "The specified plan is not a valid selection! Supported plans are: lite, standard or enterprise-3nodes-2tb."
+  }
+}
+
+variable "tags" {
+  type        = list(string)
+  description = "List of tags associated with the Event Steams instance"
+  default     = []
+}
+
+variable "region" {
+  type        = string
+  description = "IBM Cloud region where event streams will be created"
+  default     = "us-south"
+}
+
+variable "throughput" {
+  type        = number
+  description = "Throughput capacity in MB per second."
+  default     = "150" # For lite and standard plan, the allowed value of throughput is 150 MB per second. For enterprise plan it can take any value.
+}
+
+variable "storage_size" {
+  type        = number
+  description = "Storage size of the event streams in GB."
+  # For lite and standard plan, the allowed value is 2048 GB. For enterprise plan refer (https://cloud.ibm.com/docs/EventStreams?topic=EventStreams-ES_scaling_capacity#ES_storage_capacity) for allowed values.
+  # Storage capacity cannot be scaled down once instance is created.
+  # Storage capacity cannot be scaled up in lite and standard plan.
+  default = null
+}
+
+variable "service_endpoints" {
+  type        = string
+  description = "The type of service endpoint(public,private or public-and-private) to be used for connection."
+  default     = null
+  validation {
+    condition     = contains(["public", "public-and-private", "private"], var.service_endpoints)
+    error_message = "The specified service endpoint is not a valid selection! Supported options are: public, public-and-private or private."
+  }
+}
+
+variable "private_ip_allowlist" {
+  type        = string
+  description = "Range of IPs that have the access."
+  default     = null
+}
```

Comment:
I think this support needs to be replaced by CBR support. For reference, see https://github.com/terraform-ibm-modules/terraform-ibm-icd-postgresql/blob/main/variables.tf

Reply:
Looks like we still need to roll out CBR support and remove this legacy deprecated allowlisting

Reply:
I would remove this variable all together now - this is a deprecated feature. We will add CBR support in the next release anyway

Reply:
@jor2 ^^

Reply:
going to keep until we add CBR support